src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -654,6 +661,38 @@ protected virtual Task<ReturnType> InvokeCoreAsync<ReturnType>(int? id, string t
                 }
             }
         }
+        
+        private static Dictionary<string, List<string>> GetMethodToAttributeMap(object target)
+        {
+            var methodToAttributeMap = new Dictionary<string, List<string>>();
+            if (target != null)
+            {
+                for (TypeInfo t = target.GetType().GetTypeInfo(); t != null; t = t.BaseType?.GetTypeInfo())
+                {
+                    var anyMethodWithAttribute = t.DeclaredMethods.Any(m => m.GetCustomAttribute(typeof(MethodNameAttribute)) != null);
+                    if (anyMethodWithAttribute)
+                    {
+                        foreach (MethodInfo method in t.DeclaredMethods)
+                        {
+                            var attribute = method.GetCustomAttribute(typeof(MethodNameAttribute)) as MethodNameAttribute;
```

Comment:
nit: please [use the `()` cast operator](http://blog.nerdbank.net/2008/06/when-not-to-use-c-keyword.html) instead of `as` unless you expect the cast to fail in some legit circumstances. That way, we'll get more actionable exceptions if it were to ever go wrong.